### PR TITLE
Fix the time base login fail for bindings

### PR DESCRIPTION
**Pull request by marberger**
This PR holds changes that tackle the time based error, that can occur, when a binding is directly done after finishing service instance provisioning.

Therefore a retry mechanism is added, that retries the connection up to 5 times and waites a few seconds (3s) between tries.

### CODE REVIEW

**Comment by marberger** [comment]
Currently the pipeline is still failing with the same error message, but it can not be recreated locally. 
Investigating ...

**Comment by yremmet** [comment]
Changes seem good. If we're having other solutions to fix the pipeline I would like to reduce the retries again.

File: osb-service/src/main/java/de/evoila/cf/broker/custom/cassandra/CassandraBindingService.java [hunk 1]
```diff
@@ -81,6 +81,13 @@ public CassandraBindingService(BindingRepository bindingRepository, ServiceDefin
         credentialStore.createUser(serviceInstance, bindingId);
         UsernamePasswordCredential usernamePasswordCredential = credentialStore.getUser(serviceInstance, bindingId);
 
+        log.debug("##### Service User for Binding #####");
+        log.debug("Username: " + serviceInstanceUsernamePasswordCredential.getUsername());
+        log.debug("ValueName: " + CredentialConstants.SERVICE_CREDENTIALS);
+        log.debug("ServiceInstanceId: " + serviceInstance.getId());
+        log.debug("Password: " + serviceInstanceUsernamePasswordCredential.getPassword());
```
Comment (chbrinker) [inline osb-service/src/main/java/de/evoila/cf/broker/custom/cassandra/CassandraBindingService.java]: I am not that happy with logging the password. Even if it is debug log. Can we move that one to trace log?

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/CassandraDeploymentManager.java [hunk 2]
```diff
@@ -49,6 +49,19 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
             HashMap<String, Object> userProperties = adminUsers.get(0);
             UsernamePasswordCredential rootCredentials = credentialStore.createUser(serviceInstance,
                     CredentialConstants.SERVICE_CREDENTIALS, "service");
+
+            log.debug("##### Service User Creation #####");
+            log.debug("Username: " + rootCredentials.getUsername());
+            log.debug("ValueName: " + CredentialConstants.SERVICE_CREDENTIALS);
+            log.debug("ServiceInstanceId: " + serviceInstance.getId());
+            log.debug("Password: " + rootCredentials.getPassword());
```
Comment (chbrinker) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/CassandraDeploymentManager.java]: Same as above -> trace log

File: osb-service/src/main/java/de/evoila/cf/cpi/bosh/CassandraDeploymentManager.java [hunk 3]
```diff
@@ -49,6 +49,19 @@ protected void replaceParameters(ServiceInstance serviceInstance, Manifest manif
...
+            log.debug("##### After Get #####");
+            UsernamePasswordCredential tmp = credentialStore.getUser(serviceInstance, CredentialConstants.SERVICE_CREDENTIALS);
+            log.debug("Username: " + tmp.getUsername());
+            log.debug("ValueName: " + CredentialConstants.SERVICE_CREDENTIALS);
+            log.debug("ServiceInstanceId: " + serviceInstance.getId());
+            log.debug("Password: " + tmp.getPassword());
```
Comment (chbrinker) [inline osb-service/src/main/java/de/evoila/cf/cpi/bosh/CassandraDeploymentManager.java]: Same as above -> trace log

File: osb-service/src/main/java/de/evoila/cf/broker/custom/cassandra/CassandraDbService.java [hunk 4]
```diff
@@ -46,26 +60,37 @@
...
+
+        sessionBuilder.withKeyspace(keyspace)
+                .withLocalDatacenter(datacenter)
+                .withConfigLoader(getCassandraConfigHolder(username, password));
+
+        int tries = 0;
+        while (!isConnected() && tries < MAX_CONNECTION_TRIES) {
+            try {
+                session = sessionBuilder.build();
+            } catch (Exception e) {
```
Comment (chbrinker) [inline osb-service/src/main/java/de/evoila/cf/broker/custom/cassandra/CassandraDbService.java]: I am not very happy with using a general exception here. Also with using while in combination with exception raising for control flow. But do not have a better idea at the moment

**Pull request merged**